### PR TITLE
Phase4: Deploy multi-layer ServiceChain

### PR DESCRIPTION
**Pull request by iv0rish**
This source will deploy multi-layer ServiceChain to existing infrasturcture. User have to provide existing EN's security group id - ServiceChain infra will deployed, EN can communicate and make anchoring with those.

### CODE REVIEW

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+}
+
+resource "aws_security_group_rule" "ingress_l2_en_eip_network_udp" {
+  security_group_id        = aws_security_group.l2_common_sg.id
+  type                     = "ingress"
+  from_port                = 32323
+  to_port                  = 32323
+  protocol                 = "udp"
+  source_security_group_id = var.en_security_group_id
+}
```
Comment (2dvorak) [inline service-chain-aws/deploy-multi_layer-servicechain/common.tf:135]: Don't we need to open port 50505 and 50506 for `en_security_group_id`?

File: service-chain-aws/deploy-multi_layer-servicechain/terraform.tfvars [hunk 2]
```diff
@@ -0,0 +1,10 @@
+# name           = ""   # Name of every resource's name tag
+# region         = ""   # AWS region name to create all resources
+# scn_subnet_ids = ["", ""]   # A list of subnets to place SCN instance nodes. It could be better set to private subnet if it need to run without public IPs
+# spn_subnet_ids = ["", ""]   # A list of subnets to place SPN instance nodes. It usually set to public subnet contrary to scn_subnet_ids
+# sen_subnet_ids = ["", ""]   # A list of subnets to place SEN instance nodes. It usually set to public subnet contrary to scn_subnet_ids
+# en_security_group_id = ""   # A EN's security group to anchroing with the ServiceChain
+# en_subnet_ids  = ["", ""]   # A list of subnets to place EN instance nodes. It usually set to public subnet contrary to scn_subnet_ids
```
Comment (2dvorak) [inline service-chain-aws/deploy-multi_layer-servicechain/terraform.tfvars]:
```suggestion
```
We are not creating ENs, so we won't need `en_subnet_ids`, right?

**Review by 2dvorak — COMMENTED** [review]
Like I commented in #6, a README would help.
Except for that and some comments, LGTM.

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf (hunk 1 above)
Comment (2dvorak) [reply service-chain-aws/deploy-multi_layer-servicechain/common.tf:135]: On a second thought, not necessarily. I've heard that sub-bridge acts as client and connects to main-bridge port, so allowing inbound connection to main-bridge port from sub-bridge port would be sufficient. However, I added rule for inbound connection to sub-bridge port from main-bridge to sync with existing codes and reduce confusion.

Comment (2dvorak) [reply service-chain-aws/deploy-multi_layer-servicechain/common.tf:135]: Resolved in 4f1be1c

File: service-chain-aws/deploy-multi_layer-servicechain/terraform.tfvars (hunk 2 above)
Comment (2dvorak) [reply service-chain-aws/deploy-multi_layer-servicechain/terraform.tfvars]: Resolved in ef4d98c

**Review by 2dvorak — APPROVED** [review]

**Comment by 2dvorak** [comment]
@henry-will PTAL

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    values = ["machine"]
+  }
+
+  filter {
+    name   = "name"
+    values = ["CentOS Linux 7*"]
+  }
+}
+
+resource "aws_security_group_rule" "ingress_en_l2_anchoring_sg_tcp" {
```
Comment (henry-will) [inline service-chain-aws/deploy-multi_layer-servicechain/common.tf:26]: I have some error like this. Can you add an example about en_security_group_id? 

```
│ Error: InvalidGroupId.Malformed: Invalid id: "henry-sc-deploy-multi2-en" (expecting "sg-...")
│       status code: 400, request id: 29f8b94e-bcbf-408d-8939-4c709a4ac198
│
│   with aws_security_group_rule.ingress_en_l2_anchoring_sg_tcp,
│   on common.tf line 26, in resource "aws_security_group_rule" "ingress_en_l2_anchoring_sg_tcp":
│   26: resource "aws_security_group_rule" "ingress_en_l2_anchoring_sg_tcp" {
│

```

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+  security_group_id        = var.en_security_group_id
+  type                     = "ingress"
+  from_port                = 50505
+  to_port                  = 50506
+  protocol                 = "tcp"
+  source_security_group_id = aws_security_group.l2_common_sg.id
```
Comment (henry-will) [inline service-chain-aws/deploy-multi_layer-servicechain/common.tf:32]: l2_common_sg is equal to deploy-L1-L2. So when I apply different topology simultaneous, it maybe use same name. What about use different name?

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf (hunk 3 above)
Comment (2dvorak) [reply service-chain-aws/deploy-multi_layer-servicechain/common.tf:26]: You have to input the existing EN's security group ID. The security group ID is not something you can decide, but instead it is chosen by the AWS. Like the error message says, security group ID looks like `sg-0123456789abcdef`.

Suppose you have deployed L1+L2. If you want to deploy L3, you will use this terraform with `en_security_group_id` set to L2 EN's security group. The reason for requiring the parent chain EN's security group is to allow inbound connection for the bridge P2P port.

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf (hunk 4 above)
Comment (2dvorak) [reply service-chain-aws/deploy-multi_layer-servicechain/common.tf:32]: But it uses the ID and they should differ. Can you elaborate more?

File: service-chain-aws/deploy-multi_layer-servicechain/common.tf (hunk 3 above)
Comment (henry-will) [reply service-chain-aws/deploy-multi_layer-servicechain/common.tf:26]: Thanks you. I will try again and success.

**Review by henry-will — APPROVED** [review]

**Pull request merged**